src/components/Csam/CsamDetailsForm.tsx

```diff
@@ -0,0 +1,198 @@
+import { ScrollArea } from '~/components/ScrollArea/ScrollArea';
+import { Stepper, useStepperContext } from '~/components/Stepper/Stepper';
+import { Form, InputCheckboxGroup, InputRadioGroup, useForm } from '~/libs/form';
+import {
+  csamCapabilitiesDictionary,
+  csamContentsDictionary,
+  csamReportFormSchema,
+} from '~/server/schema/csam.schema';
+import { trpc } from '~/utils/trpc';
+import { isDefined } from '~/utils/type-guards';
+
+export function CsamDetailsForm() {
+  const { userId, isInternal } = useCsamContext();
+  const { next } = useStepperContext();
+  const form = useForm({
+    schema: csamReportFormSchema,
+    defaultValues: { origin: isInternal ? 'testing' : 'user' },
+    shouldUnregister: false,
+  });
+
+  const { mutate: createReport, isLoading } = trpc.csam.createReport.useMutation();
+
+  const handleSubmit = (data: z.infer<typeof csamReportFormSchema>) => {
+    createReport({
+      ...data,
+      userId,
+      modelVersionIds: useCsamModelVersionSelectStore.getState().getSelected(userId),
+      images: useCsamImageSelectStore
+        .getState()
+        .getSelected(userId)
+        .map((id) => ({ id })),
+    });
+    next();
+  };
+
+  return (
+    <ScrollArea>
+      <Container>
+        <Card>
+          <Form id="csamForm" form={form} onSubmit={handleSubmit}>
+            <Stack spacing="xl">
+              {!isInternal ? (
+                <InputRadioGroup name="minorDepiction" label="Minor depiction">
+                  <Radio value="real" label="Real" />
+                  <Radio value="non-real" label="Non-real" />
+                </InputRadioGroup>
+              ) : (
+                <InputCheckboxGroup
+                  name="capabilities"
+                  label="Model capabilities"
+                  orientation="vertical"
+                  spacing="xs"
+                >
+                  {Object.entries(csamCapabilitiesDictionary).map(([key, value]) => (
+                    <Checkbox key={key} value={key} label={value} />
+                  ))}
+                </InputCheckboxGroup>
+              )}
+
+              <InputCheckboxGroup
+                name="contents"
+                label="The images/videos in this report may involve:"
+                orientation="vertical"
+                spacing="xs"
+              >
+                {Object.entries(csamContentsDictionary).map(([key, value]) => (
+                  <Checkbox key={key} value={key} label={value} />
+                ))}
+              </InputCheckboxGroup>
+              <Input.Wrapper label="Select the resources that you'd like to have referenced in this report">
+                <ModelVersionSelectList userId={userId} />
+              </Input.Wrapper>
+              <Group position="right">
+                <Stepper.PreviousButton>Previous</Stepper.PreviousButton>
+                <Button type="submit" onClick={() => confirm('Are you sure')} loading={isLoading}>
```

Comment:
Where does `confirm` come from? 😛 
```suggestion
                <Button type="submit" onClick={() => confirm('Are you sure')} loading={isLoading}>
```

Reply:
This is the browser built in `confirm`

Reply:
I can handle this another way